src/main/java/com/workshop/demo/payload/RestaurantResponse.java

```diff
@@ -0,0 +1,45 @@
+package com.workshop.demo.payload;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import lombok.Data;
+
+@Data
+public class RestaurantResponse<T> {
```

Comment:
List<T> review should be List<Review> review as we only have one Review class, do we still need template in RestaurantResponse<T>?

From RestaurantServiceImpl it looks like the possible things we want to return in a RestaurantResponse are 
- Restaurant restaurant
- List<Restaurant> restaurant
- int score

This class could be 
```
class RestaurantResponse {
  private Restaurant restaurant;
  private List<Restaurant> restaurant;
  private int score;
  ...
}
```